pgjdbc/src/main/java/org/postgresql/core/Oid.java

```diff
@@ -84,34 +102,27 @@
    *         defined.
    */
   public static String toString(int oid) {
-    try {
-      Field[] fields = Oid.class.getFields();
-      for (Field field : fields) {
-        if (field.getInt(null) == oid) {
-          return field.getName();
-        }
-      }
-    } catch (IllegalAccessException e) {
-      // never happens
+    String name = OID_TO_NAME.get(oid);
+    if (name == null) {
+      name = "<unknown:" + oid + ">";
     }
-    return "<unknown:" + oid + ">";
+    return name;
   }
 
   public static int valueOf(String oid) throws PSQLException {
-    try {
-      return (int) Long.parseLong(oid);
-    } catch (NumberFormatException ex) {
-    }
-    try {
-      oid = oid.toUpperCase();
-      Field[] fields = Oid.class.getFields();
-      for (Field field : fields) {
-        if (field.getName().toUpperCase().equals(oid)) {
-          return field.getInt(null);
-        }
+    if (oid.length() > 0 && !Character.isDigit(oid.charAt(0))) {
+      Integer id = NAME_TO_OID.get(oid);
+      if (id == null) {
+        id = NAME_TO_OID.get(oid.toUpperCase());
+      }
+      if (id != null) {
+        return id;
+      }
+    } else {
+      try {
```

Comment:
Why parse as long and truncate to int rather than just parsing directly to int?

Reply:
That is interesting. I just replicated existing code. Makes sense to replace with Integer

Reply:
It's a `(int) Long.parseLong(oid);` because OIDs in PostgreSQL are unsigned int, using just `Integer.parseInt(oid);` will break for OIDs larger than 2147483647 with a NumberFormatException.

The cast of a long to an int simulates an unsigned int since Java don't have a primitive unsigned int, in Java 8 there is `Integer.parseUnsignedInt(oid);` but since the driver has still to support Java 6 the `(int) Long.parseLong(oid);` is the valid option.

Reply:
That is interesting